mapper/tedge_mapper/src/sm_c8y_mapper/tests.rs

```diff
@@ -0,0 +1,408 @@
+#[cfg(test)]
+#[cfg(feature = "integration-test")]
+#[cfg(feature = "mosquitto-available")]
+
+mod tests {
+    use crate::component::TEdgeComponent;
+    use mqtt_client::{Client, MqttClient, Topic, TopicFilter};
+    use once_cell::sync::OnceCell;
+    use serial_test::serial;
+    use std::io::Write;
+    use std::process::{Child, Stdio};
+    use std::time::Duration;
+    use tedge_config::{ConfigRepository, TEdgeConfig, TEdgeConfigLocation};
+
+    const MQTT_TEST_PORT: u16 = 56666;
+    static SERVER: OnceCell<Child> = OnceCell::new();
+
+    #[tokio::test]
+    #[serial]
+    async fn mapper_publishes_a_software_list_request() {
```

Comment:
We need to add cfg attribute for `mosquitto-available = []` to run only with the feature enabled for all those tests:
```suggestion
    #[tokio::test]
    #[serial]
    #[cfg_attr(not(feature = "requires-mosquitto"), ignore)]
    
    async fn mapper_publishes_a_software_list_request() {
```